crypto/nextgen_crypto/src/x25519.rs

```diff
@@ -385,3 +385,62 @@ impl<'de> de::Deserialize<'de> for X25519StaticPublicKey {
         deserializer.deserialize_bytes(X25519StaticPublicKeyVisitor {})
     }
 }
+
+//////////////////////////
+// Compatibility Traits //
+//////////////////////////
+
+/// Those transitory traits are meant to help with the progressive
+/// migration of the code base to the nextgen_crypto module and will
+/// disappear after.
+pub mod compat {
```

Comment:
Yes, this probably hints at a restructuring of the `Uniform` trait and the `KeyPair` structure as they stand in the `traits.rs` and `test_utils.rs` files respectively.

Ultimately, I think we could eliminate the `KeyPair` and have an additional method `generate_keypair` implemented for any `Uniform + PrivateKey` that uses `Uniform::generate_for_testing` and `&PrivateKey`'s `Into<PublicKey>`.